File: 1732_find-the-highest-altitude/csharp/solution.cs
public class Solution
{
    public int LargestAltitude(int[] gain)
    {
        int max = 0;
        int altitude = 0;

        foreach (int g in gain)
        {
            altitude += g;
            if (max < altitude)
                max = altitude;
        }

        return max;
    }
}

File: 1732_find-the-highest-altitude/rust/solution.rs
#[cfg(feature = "harness")]
pub struct Solution;

impl Solution {
    pub fn largest_altitude(gain: Vec<i32>) -> i32 {
        let mut max = 0;
        let mut altitude = 0;

        for g in gain {
            altitude += g;
            if max < altitude {
                max = altitude;
            }
        }

        max
    }
}

